tests/test_data/packages/small_fake_with_build_deps/backend/backend.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+import setuptools.build_meta
```

Comment:
Looks like it may be reasonable to just re-export `*` so any future and optional hooks are exposed.

Reply:
Something like below?

It would be the first occurrence of `noqa` in the code, is having this future-proofing worth establishing that precedent?

```python
from __future__ import annotations

import setuptools.build_meta

# Re-export all names in case more hooks are added in the future
from setuptools.build_meta import *  # noqa: F401, F403


def get_requires_for_build_sdist(config_settings=None):

```

Reply:
> It would be the first occurrence of `noqa` in the code, is having this future-proofing worth establishing that precedent?

Let's first establish something. Linter rules are there to keep us from (1) making accidental mistakes, (2) repeating known flaws, and (3) helping maintain consistency/conventions/etc.
Most of the time, only (3) can be detected with a 100% certainty. This effectively means that false negatives are bound to occur. This is because a machine is not a human and cannot take the semantic context into account. And so it is absolutely natural to make exceptions scoped precisely to places that are legit.

> ```python
> # Re-export all names in case more hooks are added in the future
> ```

It's usually a good idea to write a comment with the motivation next to each of such `noqa`s.

Reply:
I have added the re-export but I am not sure how to address each noqa with a separate commit.

Reply:
Marking this thread as resolved.